test/test_framework.py

```diff
@@ -605,8 +615,9 @@ def on_bar(self, event):
                 self.seen.append(f"on_bar:{type(event).__name__}:{event.handle.kind}")
                 event.defer()
 
-        pub = MyNotifier(framework, "1")
-        obs = MyObserver(framework, "1")
+        pub_a = MyNotifierA(framework, "pub_a")
+        pub_b = MyNotifierB(framework, "pub_b")
```

Comment:
I do prefer 2 types that have on events to 1 type that has 2 on events, I'm curious whether this change is just a cleanup, or whether there was something about what you've done with caching/etc that was an issue here.

Reply:
Adding the caching caused this to break because of the other issue where the `handle_kind` for `EventsBase` is hard-coded to `"on"` which caused this test to hit the "two things claiming to be..." error since the attributes were now persisted rather than dropped immediately after access.  Since we had decided previously that this pattern was unacceptable, the easiest fix was to bring this test in line with the decision that you should only ever have one `EventsBase` attribute per parent and it must be called `on`.  (The other fix would have been to make it so the `handle_kind` reflected the actual attribute name, which I'd like to do someday but was rejected as not worth it at this time.)

Reply:
right, sounds good for now